### PR TITLE
feat: myVotes filter for repo list

### LINKED ISSUE #334: Feature: Add myvotes filter - alpha branch
### Type of feature

🍕 Feature

### Current behavior

During work on https://github.com/open-sauced/hot/pull/321 I had to remove the the myVotes filter while that data s being added to the API. When it is available, we can added them back. 

### Suggested solution

Add the "votesRelation" from the future API. 

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by Deadreyo**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

The filter by "My Votes" is now functional. Filters repositories by ones that were starred by the user. Leverages the api repo's API.

## Related Tickets & Documents

closes #334 

## Mobile & Desktop Screenshots/Recordings

image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by bdougie — APPROVED** [review]
We discovered that the issue mentioned for voting was unrelated. https://github.com/open-sauced/hot/issues/385#issuecomment-1289573750.

This works as-is, so I am fine moving this to beta

File: src/hooks/useRepositoriesList.ts [hunk 1]
```diff
@@ -1,12 +1,36 @@
...
+    error.message = `${res.status} ${res.statusText}`;
+    error.stack = JSON.stringify(await res.json());
+
+    throw error;
+  }
+
+  return res.json();
+};
+
 const useRepositoriesList = (orderBy = "stars", limit = 10) => {
```
Comment (0-vortex) [inline src/hooks/useRepositoriesList.ts:26]: this is not correct - we will have 4 types of voting mechanisms, we cannot `switch` between them in a repos list; this requires a dedicated hook like `useVotedRepositories.ts`

**Review by 0-vortex — CHANGES_REQUESTED** [review]
Scope creeping with conditionals is not acceptable, however the fix is super simple :D 

We can push this forward after this small change! 🍕

File: src/hooks/useRepositoriesList.ts (hunk 1 above)
Comment (Deadreyo) [reply src/hooks/useRepositoriesList.ts]: committed

File: src/components/RepoWrap.tsx [hunk 2]
```diff
@@ -5,12 +5,14 @@ import HotRepositories from "./HotRepositories";
...
 import SecondaryNav from "./SecondaryNav";
 import { useRepositoriesList } from "../hooks/useRepositoriesList";
+import { useVotedRepositoriesList } from "../hooks/useVotedRepositoriesList";
 
 export enum RepoOrderByEnum {
   popular = "stars",
   recent = "created_at",
   upvoted = "votesCount",
   discussed = "issues",
+  myVotes = "myvotes"
```
Comment (NsdHSO) [inline src/components/RepoWrap.tsx:15]:
```suggestion
  myVotes = "myVotes"
```

Maybe camel case

File: src/hooks/useVotedRepositoriesList.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+interface PaginatedRepoResponse {
+  readonly data: DbRepo[];
+  readonly meta: PageMetaDto;
+}
+
+const apiFetcherWithToken = async (apiUrl: string) => {
+  const authSession = supabase.auth.session();
+
+  const authToken = authSession ? authSession.access_token : "";
+  const res = await fetch(`${import.meta.env.VITE_API_URL}/${apiUrl}`, { headers: { accept: "application/json", Authorization: `Bearer ${authToken}` } });
```
Comment (NsdHSO) [inline src/hooks/useVotedRepositoriesList.ts:13]: could you please give another name instead `res`

**Review by bdougie — APPROVED** [review]
Just confirming that variables seemed fixed and the preview works as expected for me

**Review by 0-vortex — APPROVED** [review]
We can move this forward, a couple of small improvement refactors but they are not critical as the other stuff mentioned in #382

**Pull request merged**